libs/cart/src/reducers/cart-billing-address/cart-billing-address.reducer.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  DaffCartBillingAddressActionTypes,
+} from '../../actions';
+import { initialState } from '../cart-initial-state';
+import { DaffCartReducerState } from '../cart-state.interface';
+import { ActionTypes } from '../action-types.type';
+import { DaffCartErrorType } from '../cart-error-type.enum';
+
+function addError(state: DaffCartReducerState, error: string) {
+  return {
+    ...state,
+    errors: {
+      ...state.errors,
+      [DaffCartErrorType.BillingAddress]: state.errors[DaffCartErrorType.BillingAddress].concat(new Array(error))
+    }
+  };
+}
+
+function resetErrors(state: DaffCartReducerState) {
+  return {
+    errors: {
+      ...state.errors,
+      [DaffCartErrorType.BillingAddress]: []
+    }
+  };
+}
+
+export function reducer(
+  state = initialState,
+  action: ActionTypes
+): DaffCartReducerState {
+  switch (action.type) {
+    case DaffCartBillingAddressActionTypes.CartBillingAddressLoadAction:
+    case DaffCartBillingAddressActionTypes.CartBillingAddressUpdateAction:
+      return { ...state, loading: true };
+
+    case DaffCartBillingAddressActionTypes.CartBillingAddressLoadSuccessAction:
+      return {
+        ...state,
+        ...resetErrors(state),
```

Comment:
Either a lot of these `resetErrors` need tested or all of them do.

Reply:
They're not being exported and their logic is tested with the reducer spec.

Reply:
I don't see this being tested in the reducer spec:
```
  describe('when CartShippingAddressLoadSuccessAction is triggered', () => {
    let result;
    let state: DaffCartReducerState;

    beforeEach(() => {
      state = {
        ...initialState,
        loading: true
      }

      const cartListLoadSuccess = new DaffCartShippingAddressLoadSuccess(cart.shipping_address);

      result = reducer(state, cartListLoadSuccess);
    });

    it('should indicate that the cart is not loading', () => {
      expect(result.loading).toEqual(false);
    });

    it('should set shipping_address from action.payload', () => {
      expect(result.cart.shipping_address).toEqual(cart.shipping_address)
    });
  });
```

Reply:
Oh you're right, my bad. I forgot to add tests covering the error reset in the first place.

Reply:
Changed in [`55e630e`](https://github.com/graycoreio/daffodil/pull/681/commits/55e630edaee54d8d47ecdd97955efe1c63ecefdd).